api/http_verbs_with_json.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"errors"
+
+	"github.com/go-resty/resty/v2"
+)
+
+func (self *ApiClient) postJSON(path string, request interface{}, response interface{}) error {
+	return self.executeJSONInJSONOut(resty.MethodPost, path, request, response)
+}
+
+func (self *ApiClient) putJSON(path string, request interface{}, response interface{}) error {
+	return self.executeJSONInJSONOut(resty.MethodPut, path, request, response)
+}
```

Comment:
Can we reuse the other verbs in a similar fashion by providing the other methods and optionally providing the other params and calling the chainable struct based on that input?